extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolProducer.java

```diff
@@ -13,43 +14,16 @@
 
     private static final Logger LOGGER = Logger.getLogger(PgPoolProducer.class);
 
-    private volatile PgPool pgPool;
-    private io.vertx.mutiny.pgclient.PgPool mutinyPgPool;
-
-    /**
-     * @deprecated The Axle API is deprecated and will be removed in the future, use {@link #mutinyPgPool} instead.
-     */
-    @Deprecated
-    private io.vertx.axle.pgclient.PgPool axlePgPool;
-    /**
-     * @deprecated The RX API is deprecated and will be removed in the future, use {@link #mutinyPgPool} instead.
-     */
-    @Deprecated
-    private io.vertx.reactivex.pgclient.PgPool rxPgPool;
-
-    void initialize(PgPool pgPool) {
-        this.pgPool = pgPool;
-    }
-
-    /**
-     * @return the <em>bare</em> PostGreSQL Pool instance.
-     */
-    @Singleton
-    @Produces
-    public PgPool pgPool() {
-        return pgPool;
-    }
+    @Inject
+    PgPool pgPool;
 
     /**
      * @return the <em>mutiny</em> PostGreSQL Pool instance. The instance is created lazily.
      */
     @Singleton
     @Produces
-    public synchronized io.vertx.mutiny.pgclient.PgPool mutinyPgPool() {
-        if (mutinyPgPool == null) {
-            mutinyPgPool = io.vertx.mutiny.pgclient.PgPool.newInstance(pgPool);
-        }
-        return mutinyPgPool;
+    public io.vertx.mutiny.pgclient.PgPool mutinyPgPool() {
+        return io.vertx.mutiny.pgclient.PgPool.newInstance(pgPool);
```

Comment:
Out of curiosity, the bean container will invoke this just once and keep a reference to the result, correct?

Reply:
Yep, it's `@Singleton` so the returned instance is stored in the context.